### ISSUE TITLE
Unbind `MenhirLib` from `Ocaml_preprocess` in `merlin-lib`

### ISSUE THREAD

**Issue by nberth** [ocaml/merlin#1533]
In version `4.6-414` the `MenhirLib` module that is bound in most `Merlin_*` modules is the one of `Ocaml_preprocess`. Is this actually intended? This could prevent reusing some parts of `Merlin_*` for other menhir-generated parsers…

As a side note, is there a plan to publish some of the scripts included here as standalone tools? I'm especially interested about `src/ocaml/preprocess/recover`.

**Comment by voodoos** [comment]
cc @let-def

**Comment by voodoos** [comment]
The intent if for Merlin not to depend on Menhir. Merlin being a tool installed in most switches it is crucial that it relies on a few dependencies as possible. Also, if we shipped the generated parser without the libs it would force users installing Merlin to have a specific version of Menhir which would be quite bad if they depend on other projects that rely on a different version of Menhir. 

There might be other workarounds to the issue you are facing. Could you tell us more about what you are planning to do ?

**Comment by nberth** [comment]
Yes I understand about the quest for having very few dependencies. My use-case, is an attempt at re-using the recovery algorithm as instantiated in `Merlin_kernel.Mreader_recover` with a grammar for another language. A direct access to some of the merlin internals would be quite helpful to achieve this.

I'll try to think about other workarounds. In the short term a separate installation with a patch as in #1534 might be fine.

**Comment by voodoos** [comment]
Closing as not planned for now. Please reopen if you feel we should think about this more.
